realm/realm-library/src/androidTest/java/io/realm/RealmRecyclerAdapterTest.java

```diff
@@ -0,0 +1,296 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class RealmRecyclerAdapterTest {
+
+    @Rule
+    public final UiThreadTestRule uiThreadTestRule = new UiThreadTestRule();
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    private Context context;
+
+    private final static String FIELD_STRING = "columnString";
+    private final static int TEST_DATA_SIZE = 47;
+
+    private boolean automaticUpdate = true;
+    private Realm testRealm;
+
+    @Before
+    public void setUp() throws Exception {
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
+        testRealm = Realm.getInstance(realmConfig);
+
+        populateTestRealm(testRealm, TEST_DATA_SIZE);
+    }
+
+    private void populateTestRealm(Realm testRealm, int objects) {
+        testRealm.beginTransaction();
+        for (int i = 0; i < objects; i++) {
+            AllTypes allTypes = testRealm.createObject(AllTypes.class);
+            allTypes.setColumnString("test data " + i);
+        }
+        testRealm.commitTransaction();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        testRealm.close();
+    }
+
+    @Test
+    public void constructor_testRecyclerAdapterParameterExceptions() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        try {
+            new RealmRecyclerAdapter(null, resultList, automaticUpdate);
+            fail("Should throw exception if context is null");
+        } catch (IllegalArgumentException ignore) {
+        }
+    }
+
+    @Test
+    @UiThreadTest
+    public void clear_testRemoveAllDataFromAdapter() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, resultList, automaticUpdate);
+        realmAdapter.clear();
+
+        assertEquals(0, realmAdapter.getItemCount());
+        assertEquals(0, resultList.size());
+    }
+
+    @Test
+    @UiThreadTest
+    public void clear_testAdapterUpdatesOnRemoveAllData() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, resultList, false);
+        RecyclerView.AdapterDataObserver adapterObserver = mock(RecyclerView.AdapterDataObserver.class);
+        realmAdapter.registerAdapterDataObserver(adapterObserver);
+        realmAdapter.clear();
```

Comment:
Great to verify this + all the ItemRange callbacks, but having a special `clear` method doesn't seem right I think. See comment below